### PR TITLE
Flyout: Remove caret, update shadow

### PR DESCRIPTION
**Pull request by dangerismycat**
## Flyout changes for Lego spec

This PR updates Flyout to the Lego spec:
- Remove caret
- Update shadow
- Update border radius

- [x] Documentation
- [x] Tests
- [ ] Experimental evidence (required for Masonry changes)
- [ ] Accessibility checkup

### CODE REVIEW

File: packages/gestalt/src/Contents.css [hunk 1]
```diff
@@ -7,7 +7,7 @@
   composes: borderBox from "./Layout.css";
   composes: rounded from "./Borders.css";
```
Comment (christianvuerings) [inline packages/gestalt/src/Contents.css]: @dangerismycat Should we also update the border radius? Was there a reason for splitting these changes out?

**Comment by dangerismycat** [comment]
The border radius change seems most easily implemented all at once, by
changing the style for `rounded`. It’s unclear how much friction there will
be on that change. These two changes seemed pretty subtle, and shouldn’t
break any current layouts, so I figured I’d put these through here and
isolate the border radius change to a separate PR.

**Comment by christianvuerings** [comment]
@dangerismycat we'll update `rounded` in a separate PR, @jennyscript is working on those. For now, please inline the style instead and no longer compose from `"./Borders.css"`

**Comment by dangerismycat** [comment]
👍

**Pull request merged**